src/features/BookSeats/ui/HallDesign.tsx

```diff
@@ -1,10 +1,115 @@
-import { FC } from 'react';
+import { FC, useEffect } from 'react';
+
+import { useQueryClient } from '@tanstack/react-query';
+import clsx from 'clsx';
+import { io } from 'socket.io-client';
+
+import { useGetHallPlan } from 'entities/hall/hooks/useGetHallPlan';
+import { RESERVATION_WEBSOCKET_URL } from 'shared/constants/api';
+import { ErrorWrapper, LoadingSpinner, Tooltip } from 'shared/ui';
+
+import { Seat } from '../types/seat';
+import { getSeatIcon } from '../utils/getSeatIcon';
+import { SeatLegend } from './SeatLegend';
 
 interface HallDesignProps {
-    hallId: string
+    hallId: string;
+    sessionId: string;
+    chosenSeats: Seat[];
+    onSeatClick: (seat: Seat) => void;
 }
 
-export const HallDesign: FC<HallDesignProps> = ({ hallId }) => {
-    console.log(hallId);
-    return <div className="w-full lg:w-2/3 flex-grow-0 flex-shrink-0 bg-black h-[750px]"></div>;
+export const HallDesign: FC<HallDesignProps> = ({
+    hallId,
+    sessionId,
+    onSeatClick,
+    chosenSeats
+}) => {
+    const { data: hallPlan, isLoading, isError } = useGetHallPlan(hallId, sessionId);
+    const queryClient = useQueryClient();
+
+    useEffect(() => {
```

Comment:
I'd suggest to create generic hook for socket logic and put it in shared
For example
useSocket which will get { socketUrl, socketEntity('reservation' in our case), onListen } as props
the idea is to hide logic of using particular library 'socket.io-client'